packages/cypress/src/plugins/plugin.ts

```diff
@@ -143,44 +144,39 @@ function buildCypressTargets(
 ) {
   const cypressConfig = getCypressConfig(configFilePath, context);
 
-  const namedInputs = getNamedInputs(projectRoot, context);
-
-  const baseTargetConfig: TargetConfiguration<CypressExecutorOptions> = {
-    executor: '@nx/cypress:cypress',
-    options: {
-      cypressConfig: configFilePath,
-    },
+  const cypressEnv = {
+    ...cypressConfig.env,
+    ...cypressConfig.e2e?.env,
   };
 
-  const targets: Record<
-    string,
-    TargetConfiguration<CypressExecutorOptions>
-  > = {};
+  const devServerTargets: Record<string, string> = cypressEnv?.devServerTargets;
 
-  if ('e2e' in cypressConfig) {
-    const e2eTargetDefaults = readTargetDefaultsForTarget(
-      options.targetName,
-      context.nxJsonConfiguration.targetDefaults,
-      '@nx/cypress:cypress'
-    );
+  const relativeConfigPath = relative(projectRoot, configFilePath);
+
+  const namedInputs = getNamedInputs(projectRoot, context);
 
+  const targets: Record<string, TargetConfiguration> = {};
+
+  if ('e2e' in cypressConfig) {
     targets[options.targetName] = {
-      ...baseTargetConfig,
+      command: `cypress run --config-file ${relativeConfigPath} --e2e --env.devServerTarget ${devServerTargets?.default}`,
```

Comment:
This is coming from cypress config's env already, do we really need to pass it again via cli?